app/assets/javascripts/application.js

```diff
@@ -20,4 +20,10 @@
 //= require nprogress
 //= require nprogress-turbolinks
 //= require main
-//= require jquery.sidr.min
\ No newline at end of file
+//= require jquery.sidr.min
+
+$(document).ajaxError(function (e, xhr, settings) {
```

Comment:
This would patch all Ajax requests, not only those created by Rails. So if a third party library that we integrate encounters a 401, we're going to end up taking the sure to the sign-in page.